### DISCUSSION TITLE
Does qdrant support binary vectors?

### DISCUSSION THREAD

**Discussion by yesletgo** [qdrant/qdrant#3825] [Ideas]
Does qdrant support the original vector being binary? If not, is there a plan to support it？

**Comment by timvisee** [comment]
At this time, no we do not.

You may be able to use [binary quantization](https://qdrant.tech/documentation/guides/quantization/#binary-quantization) to achieve a similar effect, but that would not be the same as true binary vector support.

**Reply by yesletgo** [reply]
There is a case: Many raw vectors are 64 bit binary vectors that require Hamming to calculate distance. If using qdrant binary quantization, how can we achieve this requirement? For example, how should I design my raw vectors?

**Comment by kheyer** [comment]
Throwing in a +1 for binary vector support. I use qdrant for vector search over chemical structures. Adding binary vector support would allow including Tanimoto similarity and substructure/superstructure constraints in hybrid search, which would be really useful for molecular compounds. Milvus currently supports Tanimoto/substructure/superstructure on binary vectors, but to my knowledge they don't support dense/binary hybrid search
